docs/architecture/adr-001-key-assignment.md

```diff
@@ -0,0 +1,267 @@
+# ADR 001: Key Assignment
```

Comment:
What's the purpose of this ADR, or ADRs in our repo in general? This is from a template, which is also new. Where did it come from? Different organisations have different ideas of ADRs.

The general breakdown into subheading ('On receiving a `VSCMatured`:', 'On stopping a consumer chain:' ect) is good, and the consequences and other written sections are good.

The part that is not clear to me is if the parts which are almost all code snippets are useful. I have questions. Should this be maintained? Who is the intended audience? Why would they prefer to read this instead of code? (Especially if we directed the energy away from this towards making the code more readable)?

I am still left with questions, which I believe an ADR should at least try to answer. What were the other alternative approaches to this solution that were considered and why were they not taken? What business or product goal does this solve? What security concerns arise from this feature? What testing concerns arise?

Reply:
> What's the purpose of this ADR, 

I needed a place where to add the main logic behind the key assignment design. Having a random file for each feature doesn't seem like a good practice. 

> or ADRs in our repo in general?

ADRs should describe features that are added to the repo. Unlike the spec, the ADRs should address the dependencies to Cosmos SDK and Tendermint.

>  Where did it come from? Different organisations have different ideas of ADRs.

https://github.com/cosmos/ibc-go/tree/main/docs/architecture

> Should this be maintained?

Yes. 

> Who is the intended audience?

Engineers.

> Why would they prefer to read this instead of code?

The implementation contains a lot of boiler plate code that is hard to parse for somebody not used to the SDK. We need a more concise description of a feature. 

> I am still left with questions, which I believe an ADR should at least try to answer. What were the other alternative approaches to this solution that were considered and why were they not taken?

Had we had ADRs before, every approach taken but not finalized would have been its own ADR. Or the same ADR with a commit history. Not sure that it makes sense to consumer resources to document design decisions not taken. 

> What business or product goal does this solve? What security concerns arise from this feature? What testing concerns arise?

Feel free to expand the ADR with all this info if you feel it's needed. Even better, open an issue beforehand.